thumbprint/src/main/java/com/thumbtack/thumbprint/views/edittext/ThumbprintEditTextBase.kt

```diff
@@ -65,7 +67,13 @@ abstract class ThumbprintEditTextBase(
                 ) {
                 }
 
-                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                    hasError = if (errorValidator != null) {
```

Comment:
Any thought on making hasError private and only exposing hasErrorIf
The logic would then be changed into
```hasError = errorValidator?.invoke(s) == true```

Reply:
Alternatively, instead of making the default errorValidator null, how about making it a non-null field and defaulting to a function that always returns false?

Reply:
I agree with @efujimoto 's idea, making it non-nullable and defaulted to false. 

@kevintok I think it might be useful to keep `hasError` public, in case a dev might need to block a user's interaction if any fields have errors (ie, keeping a button disabled when any fields have errors).

Reply:
Which one should we prioritize between the hasErrorIf and hasError?
We do have a separate isDisabled btw

Reply:
My disabled example was referring to enabling/disabling _other_ UI elements based on any input fields being valid or not. 

As for the `hasError`/`hasErrorIf` -- sorry, i wasn't aware this method existed here. Maybe keeping the validator nullable is good, so that it doesn't break any existing behavior.